File: app/router/ai_agent.go
package router

import (
	"github.com/aihop/gopanel/app/api"
	"github.com/aihop/gopanel/app/middleware"
	"github.com/aihop/gopanel/constant"
	"github.com/aihop/gopanel/pkg/websocket"
	"github.com/gofiber/fiber/v3"
)

func AIAgentRouter(r fiber.Router) {
	group := r.Group("ai")
	// 注意这里使用了 SUB_ADMIN 权限，因为 AI 助手也需要受到目录沙箱的限制
	group.Use(middleware.JWT(constant.UserRoleSubAdmin))
	{
		// WebSocket 端点
		group.Get("/terminal", websocket.New(api.AIAgentWsSSH))

		// Groups APIs
		group.Get("/groups", api.GetAIGroups)
		group.Post("/groups", api.CreateAIGroup)

		// Tasks APIs
		group.Get("/tasks", api.GetAITasks)
		group.Get("/tasks/:id/messages", api.GetAITaskMessages)
		group.Put("/tasks/:id", api.UpdateAITask)
		group.Delete("/tasks/:id", api.DeleteAITask)
	}
}

File: admin/src/api/modules/ai_agent.ts
import http from "@/api"
import { AIGroup, AITask, AIMessage } from "../interface/ai_agent";

// === Group APIs ===

export function getAIGroups(params: { page: number; pageSize: number }) {
  return http.get<{ items: AIGroup[]; total: number }>("/ai/groups", params)
}

export function createAIGroup(data: { name: string; description: string }) {
  return http.post<AIGroup>("/ai/groups", data)
}

// === Task APIs ===

// 获取任务列表
export function getAITasks(params: { page: number; pageSize: number; projectId?: number }) {
  return http.get<{ items: AITask[]; total: number }>("/ai/tasks", params)
}

// 获取某个任务的消息记录
export function getAITaskMessages(taskId: number) {
  return http.get<AIMessage[]>(`/ai/tasks/${taskId}/messages`)
}

// 重命名任务
export function updateAITask(taskId: number, title: string) {
  return http.put(`/ai/tasks/${taskId}`, { title })
}

// 删除任务
export function deleteAITask(taskId: number) {
  return http.delete(`/ai/tasks/${taskId}`)
}
